### PR TITLE
Networking code extraction. Mayor documentation changes. Small API changes.

### PR DESCRIPTION
**Pull request by juanignaciomolina**
# Networking code extraction. Mayor documentation changes. Small API changes.

* **Extracted networking code:** Now the networking code is placed in https://github.com/Wolox/wolmo-networking-android/pull/2
* **Mayor documentation changes:** Added Javadocs to almost every class and method in the project.
* **Small API changes:** Some small API changes were introduced to simplify their usage. This includes methods/classes names refactoring and some method's arguments modifications as well.

### CODE REVIEW

**Comment by juanignaciomolina** [comment]
I don't know why Jenkins keeps building this project. I disabled it _again_. Damn you Jenkins.

File: core/src/main/java/ar/com/wolox/wolmo/core/util/NavigationUtils.java [hunk 1]
```diff
@@ -25,85 +25,164 @@
...
-    public static void openBrowser(Context context, String url) {
+    /**
+     * Opens the browser with a given URL.
+     *
+     * @param context An instance of {@link Context}. Can't be null.
+     * @param url     The URL that the browser should open. If the URL is null or empty,
+     *                the browser will be opened with a blank page.
+     */
+    public static void openBrowser(@NonNull Context context, @Nullable String url) {
+        if (TextUtils.isEmpty(url)) url = "about:blank";
```
Comment (alvarez-fabian) [inline core/src/main/java/ar/com/wolox/wolmo/core/util/NavigationUtils.java]: I think we should move this to a constant in the class to improve readability.

File: core/src/main/java/ar/com/wolox/wolmo/core/util/NavigationUtils.java [hunk 2]
```diff
@@ -25,85 +25,164 @@
...
+     *                   the animation behaviour.
+     */
+    public static void jumpToWithAnimation(@NonNull Activity activity,
+                                           @NonNull Class clazz,
+                                           @NonNull ActivityOptionsCompat transition) {
+        ActivityCompat.startActivity(activity, new Intent(activity, clazz), transition.toBundle());
+    }
+
+    /**
+     * Sends and intent to start an {@link Activity} for the provided {@link Class} with an
```
Comment (alvarez-fabian) [inline core/src/main/java/ar/com/wolox/wolmo/core/util/NavigationUtils.java]: Typo in these javadocs. It should be "Sends an intent". :stuck_out_tongue_closed_eyes:

File: core/src/main/java/ar/com/wolox/wolmo/core/util/NavigationUtils.java [hunk 3]
```diff
@@ -25,85 +25,164 @@
...
+     * {@link String} representing the shared view in the transition.
+     *
+     * @param activity The {@link Activity} where the intent is being started.
+     * @param pairs    Paris of {@link View} and {@link String} with the shared elements.
+     * @return A non null instance of @{@link ActivityOptionsCompat}
+     */
+    @SafeVarargs
+    @NonNull
     public static ActivityOptionsCompat buildActivityOptions(
             Activity activity, Pair<View, String>... pairs) {
```
Comment (alvarez-fabian) [inline core/src/main/java/ar/com/wolox/wolmo/core/util/NavigationUtils.java]: We should add `@NonNull` here :smile:

File: core/src/main/java/ar/com/wolox/wolmo/core/util/NavigationUtils.java [hunk 4]
```diff
@@ -116,39 +195,48 @@ public Builder addSharedElement(View sharedView, String sharedString) {
             return this;
         }
 
-        public Builder addIntentObjects(IntentObject... intentObjects) {
-            for (IntentObject intentObject : intentObjects) addIntentObject(intentObject);
+        public Builder addIntentObjects(IntentExtra... intentExtras) {
+            for (IntentExtra intentExtra : intentExtras) addIntentObject(intentExtra);
```
Comment (alvarez-fabian) [inline core/src/main/java/ar/com/wolox/wolmo/core/util/NavigationUtils.java]: I think we should add curly braces to the `for`. For me it's OK to use inline `if`, but for the loops I think it's better to use braces. You can keep it like this if you want. :smile:

File: core/src/main/java/ar/com/wolox/wolmo/core/util/ToastUtils.java [hunk 5]
```diff
@@ -23,12 +23,46 @@
...
+ */
 public class ToastUtils {
-    public static void showToast(int resId) {
+
+    /**
+     * Displays a text message from a resource ID inside a {@link Toast}, briefly
+     *
+     * @param resId A resource ID from a {@link String} with the message to be displayed
+     */
+    public static void show(int resId) {
```
Comment (alvarez-fabian) [inline core/src/main/java/ar/com/wolox/wolmo/core/util/ToastUtils.java]: We can add `@IdRes` here.

File: core/src/main/java/ar/com/wolox/wolmo/core/util/ToastUtils.java [hunk 6]
```diff
@@ -23,12 +23,46 @@
...
+        Toast.makeText(ContextUtils.getAppContext(), message, Toast.LENGTH_SHORT).show();
+    }
+
+    /**
+     * Displays a text message from a resource ID inside a toast, during a longer than usual
+     * interval
+     *
+     * @param resId A resource ID from a {@link String} with the message to be displayed
+     */
+    public static void showLong(int resId) {
```
Comment (alvarez-fabian) [inline core/src/main/java/ar/com/wolox/wolmo/core/util/ToastUtils.java]: Here too :smile:

File: core/src/main/java/ar/com/wolox/wolmo/core/util/ToastUtils.java (hunk 5 above)
Comment (juanignaciomolina) [inline core/src/main/java/ar/com/wolox/wolmo/core/util/ToastUtils.java]: Good idea, in fact, I added a `@StringRes` annotation which enforces the use of a String identifier (not just any identifier).

File: core/src/main/java/ar/com/wolox/wolmo/core/util/NavigationUtils.java (hunk 4 above)
Comment (juanignaciomolina) [inline core/src/main/java/ar/com/wolox/wolmo/core/util/NavigationUtils.java]: Sure, I agree.

File: core/src/main/java/ar/com/wolox/wolmo/core/util/NavigationUtils.java (hunk 3 above)
Comment (juanignaciomolina) [inline core/src/main/java/ar/com/wolox/wolmo/core/util/NavigationUtils.java]: Good catch!

File: core/src/main/java/ar/com/wolox/wolmo/core/util/ToastUtils.java (hunk 6 above)
Comment (juanignaciomolina) [inline core/src/main/java/ar/com/wolox/wolmo/core/util/ToastUtils.java]: Added `@StringRes` annotation here.

File: core/src/main/java/ar/com/wolox/wolmo/core/util/NavigationUtils.java (hunk 1 above)
Comment (juanignaciomolina) [inline core/src/main/java/ar/com/wolox/wolmo/core/util/NavigationUtils.java]: Ok, done.

File: core/src/main/java/ar/com/wolox/wolmo/core/util/NavigationUtils.java (hunk 2 above)
Comment (juanignaciomolina) [inline core/src/main/java/ar/com/wolox/wolmo/core/util/NavigationUtils.java]: Man, reviewing the Javadocs is next gen reviewing. Good job!

giphy-downsized

**Pull request merged**